src/odemis/driver/technolution.py

```diff
@@ -659,100 +672,151 @@ def _getScanStepSizeBits(self):
 
-        # Scanning time which does not fall in a multiple of whole descan periods
+        # Remainder of the scanning time which is not a whole multiple of the descan periods.
         remainder_scanning_time = scanning_time % descan_period
 
-        # Use timing setpoints because the maximum is defined by the time scale.
-        time_setpoints_scan = numpy.linspace(0, scanning_time - remainder_scanning_time,
-                                             int(scanning_time // descan_period))
-        # TODO K.K. check precision of descan period on ASM with Robert
+        # Calculate the time stamps for the setpoints excluding the remainder of the scanning time.
+        number_setpoints = scanning_time // descan_period
+        timestamp_setpoints = numpy.linspace(0, scanning_time - remainder_scanning_time, number_setpoints)
+
         # An absolute tolerance of 1e-6 is one order of magnitude more precision than the descan period.
+        # Use almost_equal to handle floating point errors.
         if not all(almost_equal(time_step, descan_period, rtol=0, atol=1e-6) for time_step in
-                   numpy.unique(numpy.diff(time_setpoints_scan))):
+                   numpy.unique(numpy.diff(timestamp_setpoints))):
             # If raised likely caused by incorrect input time_setpoints_scan in linspace
-            raise ValueError("Incorrect timing between the setpoints has been found for acquisition in X direction")
+            raise ValueError("Incorrect timing between the setpoints has been found for acquisition in x direction")
 
-        # Slope of the scan gain in bits per second
-        scan_gain_slope = (scan_gain_bits - scan_offset_bits) / scanning_time
-        scanning_points = scan_gain_slope * time_setpoints_scan + scan_offset_bits  # x_setpoints=slope*time + offset
+        # Slope of the scanning ramp in bits per second
+        scanning_slope = (scan_gain_bits - scan_offset_bits) / scanning_time
+        scanning_points = scanning_slope * timestamp_setpoints + scan_offset_bits  # x_setpoints=slope*time + offset
 
+        # Use almost_equal to handle floating point errors.
         if not almost_equal(remainder_scanning_time, 0, rtol=0, atol=1e-10):
             # Allow the descanner to finish scanning the last pixels by adding one setpoint with the scan gain value.
             scanning_points = numpy.hstack((scanning_points, scan_gain_bits))
-            # Extra setpoint can be added at the maximum entered value (scan_gain) which is approved by Wilco
-            # and Andries. Because of the inertia of the mirror it is expected that this is the best solution
-            # (relatively fast and safe)
+            # Note: For now the extra setpoint to compensate for the remaining scanning time has a value equal to the
+            # scan gain. Because of the inertia of the mirror it is expected that this is a decent solution. (
+            # relatively fast and safe)
```

Comment:
Replace with what we discussed together. The second sentence is still the old one.

Reply:
We did discuss this lets find it out.